File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count=1
        val=nums[0]

        for i in range(1,len(nums)):
            if nums[i]==val:
                count=count+1
            elif count==0:
                count=1
                val=nums[i]
            else:
                count=count-1
        
        return val

File: Data Structures & Algorithms/concatenation-of-array/submission-2.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n2=2*(nums.size());
        vector<int> ans(n2);
        int n=nums.size();
        for(int i=0;i<2*n;i++){
            if(i>=n){
                ans[i]=nums[i-n];
            }
            else{
                ans[i]=nums[i];
            }
        }
        return ans;
    }
};